[Issue] image export panics with entry in nil map

<!--
Note: Please search to see if an issue already exists for the bug you encountered.
-->

### Current Behavior

```
$ regctl image export localhost:5000/library/alpine:latest alpine.tar
panic: assignment to entry in nil map

goroutine 1 [running]:                              
github.com/regclient/regclient.(*RegClient).ImageExport(0xc000169810, {0x8bfcb8, 0xc000102678}, {{0x81f5e2, 0x3}, {0x0, 0x0}, {0x7ffeb3d212ea, 0xe}, {0x7ffeb3d212f9, ...}, ...}, ...)
        /home/bmitch/data/git/regclient/image.go:394 +0x58b
main.runImageExport(0xb4af20, {0xc0000c4a40, 0x2, 0x2})
        /home/bmitch/data/git/regclient/cmd/regctl/image.go:202 +0x3bf
github.com/spf13/cobra.(*Command).execute(0xb4af20, {0xc0000c49e0, 0x2, 0x2})
        /home/bmitch/data/git/regclient/vendor/github.com/spf13/cobra/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xb4d4a0) 
        /home/bmitch/data/git/regclient/vendor/github.com/spf13/cobra/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
        /home/bmitch/data/git/regclient/vendor/github.com/spf13/cobra/command.go:902
main.main()                                         
        /home/bmitch/data/git/regclient/cmd/regctl/main.go:9 +0x25
```

<!-- A concise description of what you're experiencing. -->

### Expected Behavior

The `alpine.tar` should be created.
<!-- A concise description of what you expected to happen. -->

### Steps To Reproduce

Pick an image and export it with

```
$ regctl image export <image-name> <filename>
```

<!--
Example: steps to reproduce the behavior:
1. In this environment...
2. With this config...
3. Run '...'
4. See error...
-->

### Version

```text
$ regctl version
{
  "VCSRef": "55ebfec3b3a82c0de275356d1b0d653ddd597948",
  "VCSTag": ""
}
```

### Environment

<!--
Please include:
- How you are running the tool: downloaded binary or from within a container
- Your platform: Linux, Windows, or Mac, x86 or ARM.
- Your registry: Docker Hub, ECR, GCR, ACR, Harbor, registry:2, etc.
-->

- Running as binary or container: binary
- Host platform: linux
- Registry description: distribution/distribution

### Anything else

<!--
Links? References? Anything that will give us more context about the issue that you are encountering!
-->

--- a/image.go
+++ b/image.go
@@ -391,12 +391,15 @@ func (rc *RegClient) ImageExport(ctx context.Context, ref ref.Ref, outStream io.
 
 	// create a manifest descriptor
 	mDesc := m.GetDescriptor()
+	if mDesc.Annotations == nil {
+		mDesc.Annotations = map[string]string{}
+	}
 	mDesc.Annotations[annotationImageName] = ref.CommonName()
 	mDesc.Annotations[annotationRefName] = ref.Tag
 
 	// generate/write an OCI index
-	ociIndex.SchemaVersion = 2
-	ociIndex.Manifests = append(ociIndex.Manifests, mDesc) // add the descriptor to the manifest list
+	ociIndex.Versioned = v1.IndexSchemaVersion
+	ociIndex.Manifests = []types.Descriptor{mDesc} // initialize with the descriptor to the manifest list
 	err = twd.tarWriteFileJSON(ociIndexFilename, ociIndex)
 	if err != nil {
 		return err

